src/main/java/transforms/PaddingTransform.java

```diff
@@ -0,0 +1,114 @@
+import java.util.regex.Matcher;
+
+public class PaddingTransform extends Transformation<Map<String, String>> {
+    private final String pattern;
+    private final int length;
+    private final boolean pre;
+    private final boolean cutoff;
+
+    /**
+     * A transform that pads the input string to a certain length by appending or prepending a repeating pattern
+     *
+     * @param pattern:            pad will be constructed by repeating this string
+     * @param length:             target length of resulting string
+     * @param pre:                whether to prepend (as opposed to append)
+     * @param cutoff:             whether to trim the string to the target size if it's larger
+     *
+     */
+    @JsonCreator
+    public PaddingTransform(
+            @JsonProperty( Constants.PATTERN ) String pattern,
+            @JsonProperty( Constants.LENGTH ) int length,
+            @JsonProperty( Constants.PRE ) boolean pre,
+            @JsonProperty( Constants.CUTOFF ) boolean cutoff
+            ) {
+        this.pattern = pattern;
+        this.length = length;
+        this.pre = pre;
+        this.cutoff = cutoff;
+    }
+
+    @JsonCreator
+    public PaddingTransform(
+            @JsonProperty( Constants.PATTERN ) String pattern,
+            @JsonProperty( Constants.LENGTH ) int length,
+            @JsonProperty( Constants.PRE ) boolean pre
+    ) {
+        this.pattern = pattern;
+        this.length = length;
+        this.pre = pre;
+        this.cutoff = false;
+    }
+
+
+    @JsonProperty( Constants.PATTERN )
+    public String getPattern() {
+        return pattern;
+    }
+
+    @JsonProperty( Constants.LENGTH )
+    public int getLength() {
+        return length;
+    }
+
+    @JsonProperty( Constants.PRE )
+    public boolean getPre() {
+        return pre;
+    }
+
+    @JsonProperty( Constants.CUTOFF )
+    public boolean getCutoff() {
+        return cutoff;
+    }
+
+    @Override
+    public Object applyValue( String o ) {
+
+        if (length < 0) {
+            throw new IllegalStateException( "Negative length given." );
+        }
+
+        if (pattern.equals("")) {
+            throw new IllegalStateException( "Empty pattern given." );
+        }
+
+        if (o.length() > length) {
+            if (cutoff) {
+                if (pre) {
+                    return o.substring(o.length() - length);
+                }
+                else {
+                    return o.substring(0, length);
+                }
+            }
+            return o;
+        }
+
+        StringBuilder builder = new StringBuilder(o);
+        if (pre) {
+            while (length > builder.length()) {
+                builder.insert(0,pattern);
+            }
+            return builder.substring(builder.length() - length);
+        }
+        else {
+            while (length > builder.length()) {
+                builder.append(pattern);
+            }
+            return builder.substring(0, length);
```

Comment:
So I believe this is a no-op because `substring(0, length)` is basically just "return the whole string". So you'll probably just want `builder.toString()` here

Reply:
In this case length <= builder.length(). Necessary in the case where length is not an integer multiple of the pattern length.

Reply:
Oohhhh I see. Didn't realize this class also did trimming. Cool!